Safety/Drivers/Peripherals/Src/interchip_comm_S.c

```diff
@@ -0,0 +1,23 @@
+#include "interchip_comm_S.h"
+#include "debug.h"
+#include <stdio.h>
+#include "spi.h"
+
+
+Interchip_StoA_Packet* dataTX;
+Interchip_AtoS_Packet* dataRX;
+
+HAL_StatusTypeDef transmit_status;
+
+void interchipInit(Interchip_StoA_Packet* ptrTX, Interchip_AtoS_Packet* ptrRX){
+    dataTX = ptrTX;
+    dataRX = ptrRX;
+    //start TxRx
+    HAL_SPI_TransmitReceive_IT(&hspi1,(uint8_t *)dataTX,(uint8_t *)dataRX, sizeof(Interchip_AtoS_Packet)/sizeof(uint16_t));
+}
+
+void HAL_SPI_TxRxCpltCallback(SPI_HandleTypeDef *hspi)
+{
+  //Once packet recieved, listen for next packet
+  HAL_SPI_TransmitReceive_IT(&hspi1,(uint8_t *)dataTX,(uint8_t *)dataRX, sizeof(Interchip_AtoS_Packet)/sizeof(uint16_t));
```

Comment:
Should include checks for which SPI instance is generating the callback. i.e.
`if (hspi->Instance == SPI1) {
HAL_SPI...
}`

Reply:
Yep, forgot to add that in. Shouldn't matter because there is only 1 SPI, but will fix.